components/post_markdown/system_message_helpers.jsx

```diff
@@ -16,15 +16,15 @@ import CombinedSystemMessage from 'components/post_view/combined_system_message'
 import PostAddChannelMember from 'components/post_view/post_add_channel_member';
 
 function renderUsername(value) {
-    return renderFormattedText(value, {markdown: false});
-}
+    const username = (value[0] === '@') ? value : `@${value}`;
 
-function renderUsernameForUserIdAndUsername(userId, username) {
-    const displayUsername = Utils.getDisplayNameByUserId(userId);
-    if (displayUsername && displayUsername.trim() !== '') {
-        return renderUsername(displayUsername);
-    }
-    return renderUsername(username);
+    const options = {
+        mentionKeys: [{key: username}],
```

Comment:
Is it necessary to pass `mentionKeys` and `mentionHighlight` here? The rest of this looks good, but I don't think those are needed